pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -30,15 +34,17 @@ def __init__(self, trainer):
         self.max_batches = None
         self.warning_cache = WarningCache()
         self.num_dataloaders = None
+        self.batch_indices = None
 
-    def on_trainer_init(self):
+    def on_trainer_init(self, replace_batch_sampler_auto_id: bool):
```

Comment:
is this the right place to set this attribute? no other argument is passed :thinking:

Reply:
This attribute is used evaluation_loop to transform testing batch_samplers. I think it is fine.